server/issue_test.go

```diff
@@ -279,6 +280,255 @@ func TestRouteShareIssuePublicly(t *testing.T) {
 	}
 }
 
+func TestShouldReceiveNotification(t *testing.T) {
+	cs := ConnectionSettings{}
+	cs.RolesForDMNotification = make(map[string]bool)
+	cs.RolesForDMNotification[subCommandAssignee] = true
+	cs.RolesForDMNotification[subCommandMention] = true
+	tests := map[string]struct {
+		role                   string
+		notification           bool
+		RolesForDMNotification map[string]bool
+	}{
+		subCommandAssignee: {
+			role:         subCommandAssignee,
+			notification: true,
+		},
+		subCommandMention: {
+			role:         subCommandMention,
+			notification: true,
+		},
+		subCommandReporter: {
+			role:         subCommandReporter,
+			notification: false,
+		},
+		subCommandWatching: {
+			role:         subCommandWatching,
+			notification: false,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			val := cs.ShouldReceiveNotification(tt.role)
+			assert.Equal(t, tt.notification, val)
+		})
+	}
+
+}
+
+func TestFetchConnectedUser(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+
+	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	p := &Plugin{}
+	p.SetAPI(api)
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+	p.userStore = getMockUserStoreKV()
+	tests := map[string]struct {
+		instanceID  types.ID
+		client      Client
+		connection  *Connection
+		wh          webhook
+		expectedErr error
+	}{
+		"Issue Feild not found": {
+			instanceID: testInstance1.InstanceID,
+			client:     nil,
+			connection: nil,
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{},
+				},
+			},
+			expectedErr: nil,
+		},
+		"Unable to load instance": {
+			instanceID: "instanceID",
+			client:     nil,
```

Comment:
both client and connection is nil in this case and the other as well have a test case where one is nil